esp-wifi/src/wifi/mod.rs

```diff
@@ -2620,7 +2620,14 @@ pub fn new<'d>(
             esp_wifi_result!(esp_wifi_set_country(&country))?;
         }
 
-        controller.set_power_saving(PowerSaveMode::default())?;
+        #[cfg(initial_power_save_mode_none)]
+        const INITIAL_POWER_SAVE_MODE: PowerSaveMode = PowerSaveMode::None;
+        #[cfg(initial_power_save_mode_min)]
+        const INITIAL_POWER_SAVE_MODE: PowerSaveMode = PowerSaveMode::Minimum;
+        #[cfg(initial_power_save_mode_max)]
+        const INITIAL_POWER_SAVE_MODE: PowerSaveMode = PowerSaveMode::Maximum;
+
+        controller.set_power_saving(INITIAL_POWER_SAVE_MODE)?;
```

Comment:
This AGAIN makes it so that PowerSaveMode::default() is not the default powersave mode. Why are you intentionally implementing it the wrong way...

Reply:
we should remove Default from PowerSaveMode or do you suggest to have options "none","min","max" and "default"

Reply:
if anything the configs should change the value of PowerSaveMode::default(). But I think I'd prefer just turning these into runtime config options, that we pass in to `new()`

Reply:
> But I think I'd prefer just turning these into runtime config options, that we pass in to new() using a config struct.

That at least is aligned with what we do in other places - so I'd say that's the right thing to do